synapse/config/jwt_config.py

```diff
@@ -45,10 +45,34 @@ def read_config(self, config, **kwargs):
 
     def generate_config_section(self, **kwargs):
         return """\
-        # The JWT needs to contain a globally unique "sub" (subject) claim.
+        # JSON web token integration. The following settings can be used to make
+        # Synapse JSON web tokens for authentication, instead of its internal
+        # password database.
+        #
+        # Each JSON Web Token needs to contain a "sub" (subject) claim, which is
+        # used as the localpart of the mxid.
```

Comment:
There's no guarantee the `sub` will meet mxid character restrictions. How do we map under this scenario?

Reply:
Currently it is assumed that it meets the restrictions.

Reply:
I looked at this again since I wasn't 100% sure this was true and the string just gets jammed into one of our `UserID` objects:

https://github.com/matrix-org/synapse/blob/21a212f8e50343e9b55944fa75ece7911fd2cb70/synapse/rest/client/v1/login.py#L376-L380

So yeah, this is essentially completely unchecked. 😢